.github/workflows/ci.yml

```diff
@@ -197,10 +197,25 @@ jobs:
             done
           done
 
+  ctc-check:
+    name: Obtain CTC Lock
+    if: github.ref_type == 'tag'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Obtain CTC Lock via TPS API
+        run: |
+          curl -sS -X PUT \
```

Comment:
@joshwlewis This check currently won't actually block releases, since CTC uses non-HTTP-200 responses to denote blocking a lock (https://github.com/heroku/tps/blob/master/docs/ctc.md#responses), and Curl ignores these unless the `--fail` argument is used.

eg compare to the Cheverny implementation, which checks for HTTP 200/201:
https://github.com/heroku/cheverny/blob/f27b132b8b829b16d2161977f00316faf0f06520/lib/mediators/stack/ctc_checker.rb#L27

For bonus points, we should probably also use `--fail-with-body` rather than `--fail`, since the former will also print the API response message:
https://curl.se/docs/manpage.html#--fail-with-body

Reply:
For even more bonus points, perhaps we should add `--connect-timeout` and `--retry` to reduce flakiness?